src/orion/plotting/backend_plotly.py

```diff
@@ -549,6 +549,318 @@ def build_frame():
+            ] * len(df)
+            df["order"] = range(len(df))
+            frames.append(df)
+
+        return pd.concat(frames)
+
+    def get_objective_name(experiments):
+        """Infer name of objective based on trials of one experiment"""
+        if compute_average and isinstance(experiments, dict):
+            return get_objective_name(sum(map(list, experiments.values()), []))
+
+        if isinstance(experiments, dict):
+            experiments = experiments.values()
+
+        for experiment in experiments:
+            trials = experiment.fetch_trials_by_status("completed")
+            if trials:
+                return trials[0].objective.name
+        return "objective"
+
+    ORDER_KEYS = ["suggested", "reserved", "completed"]
+
+    if not experiments:
+        raise ValueError("Parameter 'experiment' is None")
+
+    if order_by not in ORDER_KEYS:
+        raise ValueError(f"Parameter 'order_by' is not one of {ORDER_KEYS}")
+
+    df = build_groups()
+    fig = go.Figure()
+
+    if df.empty:
+        return fig
+
+    names = set(df["name"])
+    for i, name in enumerate(sorted(names)):
+        exp_data = df[df["name"] == name]
+        if "best_mean" in exp_data:
+            x = exp_data["duration_mean"]
+            y = exp_data["best_mean"]
+        else:
+            x = exp_data["duration"]
+            y = exp_data["best"]
+
+        fig.add_trace(
+            go.Scatter(
+                x=x,
+                y=y,
+                visible=True,
+                mode="lines+markers",
+                line=dict(color=px.colors.qualitative.G10[i]),
+                name=name,
+                legendgroup=name,
+            )
+        )
+        fig.add_trace(
+            go.Scatter(
+                x=x / 60.0,
+                y=y,
+                visible=False,
+                mode="lines+markers",
+                line=dict(color=px.colors.qualitative.G10[i]),
+                name=name,
+                legendgroup=name,
+            )
+        )
+        fig.add_trace(
+            go.Scatter(
+                x=x / (60.0 * 60.0),
+                y=y,
+                visible=False,
+                mode="lines+markers",
+                line=dict(color=px.colors.qualitative.G10[i]),
+                name=name,
+                legendgroup=name,
+            )
+        )
+        fig.add_trace(
+            go.Scatter(
+                x=x / (60.0 * 60.0 * 60.0),
+                y=y,
+                visible=False,
+                mode="lines+markers",
+                line=dict(color=px.colors.qualitative.G10[i]),
+                name=name,
+                legendgroup=name,
+            )
+        )
+
+    fig.update_layout(
+        updatemenus=[
+            dict(
+                type="buttons",
+                direction="right",
+                active=0,
+                x=0.57,
+                y=1.2,
+                buttons=list(
```

Comment:
This is cool! 

We could have a leaner version however, without requiring any intervention from the user. 

if max_duration > 5 days: format in days
elif max_duration > 5 hours: format in hours
elif max_duration > 5 minutes: format in minutes
else: format in seconds.

Reply:
Done in 060b747. Thanks!